src/Storages/MergeTree/MutateTask.cpp

```diff
@@ -85,6 +86,41 @@ static void splitAndModifyMutationCommands(
 
                 if (command.type == MutationCommand::Type::MATERIALIZE_COLUMN)
                     mutated_columns.emplace(command.column_name);
+
+                if (command.type == MutationCommand::Type::MATERIALIZE_INDEX)
```

Comment:
I thought there can be some noop materialization of indices and projections (unchanged). In that case we should not read dependent columns.

Reply:
This code is needed in case of when `isFullPartStorage(part) == false` which is not a case now. I've removed this code.